hadoop-plugin/src/main/groovy/com/linkedin/gradle/dependency/CheckDependencyTask.groovy

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.gradle.dependency;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.tasks.TaskAction;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Unmarshaller;
+
+class CheckDependencyTask extends DefaultTask {
+
+  Map<Dependency, List<DependencyPattern>> matchedDependencyPatterns;
+  String DEPENDENCY_PATTERN_FILE = "sample-dependency-pattern.xml";
```

Comment:
I feel a file with gradle notations or something similar is much easier to specify